packages/contracts-ops/contracts/CallBatch.sol

```diff
@@ -0,0 +1,117 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity 0.7.6;
+
+// solhint-disable quotes
+
+import {GovernanceMessage} from "@nomad-xyz/contracts-core/contracts/governance/GovernanceMessage.sol";
+import {TypeCasts} from "@nomad-xyz/contracts-core/contracts/XAppConnectionManager.sol";
+
+import "forge-std/Script.sol";
+
+abstract contract CallBatch is Script {
+    GovernanceMessage.Call[] public calls;
+
+    bool public complete;
+    string public domain;
+    string public outputFile;
+
+    function initializeCallBatch(
+        string memory _domain,
+        string memory _outputFile
+    ) public {
+        require(bytes(domain).length == 0, "already initialized");
+        require(bytes(outputFile).length == 0, "already initialized");
+        domain = _domain;
+        outputFile = string(abi.encodePacked("./actions/", _outputFile));
+    }
+
+    function push(address to, bytes memory data) public {
+        push(TypeCasts.addressToBytes32(to), data);
```

Comment:
add
```
        require(!complete, "callbatch has been completed");
```
?

Reply:
no, this is required in the inner push call